### PR TITLE
[2201.9.0] Fix adding namespaces when the option is not selected

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
$subject.

Fixes: https://github.com/wso2/ballerina-vscode/issues/560

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/xml-to-record-converter/src/test/resources/ballerina/sample_20.bal [hunk 1]
```diff
@@ -23,7 +23,7 @@ type Genre_Genre record {
 };
 
 type Currency_Price record {
-    string \#content;
+    decimal \#content;
```
Comment (nipunayf) [inline misc/xml-to-record-converter/src/test/resources/ballerina/sample_20.bal:26]: Is this relevant to this PR?

Comment (mindula) [reply misc/xml-to-record-converter/src/test/resources/ballerina/sample_20.bal:26]: Yes this fix should be added as well

File: misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java [hunk 2]
```diff
@@ -277,9 +277,9 @@ private static List<Node> getRecordFieldsForXMLElement(Element xmlElement, boole
         for (int i = 0; i < xmlAttributesMap.getLength(); i++) {
             org.w3c.dom.Node xmlNode = xmlAttributesMap.item(i);
             if (xmlNode.getNodeType() == org.w3c.dom.Node.ATTRIBUTE_NODE) {
-                if ((xmlNode.getPrefix() == null &&
-                        XMLNS_PREFIX.equals(xmlNode.getLocalName())) || (XMLNS_PREFIX.equals(xmlNode.getPrefix()) &&
-                                xmlNode.getLocalName().equals(xmlElement.getPrefix())) && withNameSpace) {
+                if (((xmlNode.getPrefix() == null && XMLNS_PREFIX.equals(xmlNode.getLocalName())) ||
```
Comment (prakanth97) [inline misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java:280]: This may not relate to this fix, but check why we have this `xmlNode.getPrefix() == null && XMLNS_PREFIX.equals(xmlNode.getLocalName()` check? it seem it always false to me.

Comment (mindula) [reply misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java:280]: If we consider the below scenario, the `xmlNode` does not have a prefix and the `localName` of the `xmlNode` is `xmlns`. Hence, this condition will be true.

```
<library xmlns="http://example.com/library">
    <book>
        <title>The Great Gatsby</title>
        <author>F. Scott Fitzgerald</author>
        <genre>Drama</genre>
    </book>
</library>
```

**Review by prakanth97 — APPROVED** [review]
LGTM

**Review by KavinduZoysa — APPROVED** [review]
